File: simp_regression.r
# load the datset
datset = read.csv('Salary_Data.csv')

library('caTools')
set.seed(123)
split = sample.split(datset$Salary, SplitRatio = 2/3)
train_set = subset(datset, split == TRUE)
test_set = subset(datset, split == FALSE)

# fitting simple linear regression to train_set 
regressor = lm(formula = Salary ~ YearsExperience, data = train_set)

# prdict test_set results
y_pred = predict(regressor, newdata = test_set)

# visualising the train_set
# install.packages('ggplot2')
library('ggplot2')
ggplot() +
  geom_point(aes(x = train_set$YearsExperience, y = train_set$Salary),
             colour = 'red') +
  geom_line(aes(x = train_set$YearsExperience, y = predict(regressor, newdata = train_set)),
             colour = 'blue') +
  ggtitle('Salary vs Experience (Training set)') + 
  xlab('Years of Experience') +
  ylab('Salary') 

# visualising the train_set
# library('ggplot2')
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'black') + 
  geom_line(aes(x = train_set$YearsExperience, y = predict(regressor, newdata = train_set)),
            colour = 'blue') + 
  ggtitle('Salary vs Experience (Test set)') + 
  xlab('Years of experience') + 
  ylab('Salary')

File: simp_regression.py
# -*- coding: utf-8 -*-
"""
Created on Sat Nov  9 13:39:36 2019

@author: vhs
"""

#importing the libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#importing the dataset
datset = pd.read_csv('Salary_Data.csv')
x = datset.iloc[:,:-1].values
y = datset.iloc[:,-1].values

#splitting the dataset into training and test set
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 1/3, random_state =0)

#fitting linear regression to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(x_train,y_train)

#predicting the test set results
y_pred = regressor.predict(x_test)

#visualizing the training set results
plt.scatter(x_train, y_train, color = 'red')
plt.plot(x_train, regressor.predict(x_train), color ='blue')
plt.title('Salary vs Experience (Training set)')
plt.xlabel('Years of experirnce')
plt.ylabel('Salary')
plt.show()

#visualizing the test set results
plt.scatter(x_test, y_test, color = 'yellow')
#plt.scatter(x_test,y_pred, color = 'orange')
plt.plot(x_train, regressor.predict(x_train),color = 'blue')
plt.title('Salary vs Experirence (test set)')
plt.xlabel('Years of experirnce')
plt.ylabel('Salary')
plt.show()